fedimint-client/src/lib.rs

```diff
@@ -533,50 +565,143 @@ const SUPPORTED_CORE_API_VERSIONS: &[fedimint_core::module::ApiVersion] =
 
 pub type ModuleGlobalContextGen = ContextGen<DynGlobalClientContext>;
 
+/// Resources particular to a module instance
+pub struct ClientModuleInstance {
+    /// Instance id of the module
+    pub id: ModuleInstanceId,
+    /// Module-specific DB
+    pub db: Database,
+    /// Module-specific API
+    pub api: DynModuleApi,
+}
+
+pub struct Client {
```

Comment:
`ClientInner` was not pub, but `Client` is now pub.. how come?

Reply:
Now we need to be able to call methods on it directly, as `&Client` supposed to be usable, where before `Client` (now `ClientArc`) had all the public methods, usually just delegating to the inner one.

We can achieve the same control of method visibility with `pub`, `pub(crate)` and private methods.